### PR TITLE
Fix #1943: Open staff window on muliplayer clients when hiring new staff instead of ramdom visitor

### LINKED ISSUE #1943: Placing Handyman/Security Guard/Mechanic In Multiplayer Causes Crash
Placing a new worker in Multiplayer causes the game to crash the first time, subsequently it does work, albeit buggy.

### PR DESCRIPTION
**Pull request by manuelVo**

### CODE REVIEW

File: src/peep/staff.c [hunk 1]
```diff
@@ -327,6 +327,20 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
 									   edi);
 }
 
+void game_command_callback_hire_new_staff_member(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp)
+{
+    int sprite_index = edi;
+    if (sprite_index == 0xFFFF)
```
Comment (IntelOrca) [inline src/peep/staff.c]: Use `SPRITE_INDEX_NULL`

File: src/peep/staff.c [hunk 2]
```diff
@@ -327,6 +327,20 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
...
+    {
+        rct_window *window = window_find_by_class(WC_STAFF_LIST);
+        window_invalidate(window);
+    }
+    else
+    {
+        window_staff_open(&get_sprite(sprite_index)->peep);
```
Comment (IntelOrca) [inline src/peep/staff.c]: I encourage that you pull expressions out into locals as it makes it much easier to debug and diagnose problems.
```
rct_peep * peep = &get_sprite(sprite_index)->peep;
window_staff_open(peep);
```

**Comment by manuelVo** [comment]
Done

File: src/peep/staff.c [hunk 3]
```diff
@@ -327,6 +327,21 @@ void game_command_hire_new_staff_member(int* eax, int* ebx, int* ecx, int* edx,
...
+        rct_window *window = window_find_by_class(WC_STAFF_LIST);
+        window_invalidate(window);
+    }
+    else
+    {
+        rct_peep *peep = &get_sprite(sprite_index)->peep;
+        window_staff_open(peep);
+    }
+}
+
```
Comment (IntelOrca) [inline src/peep/staff.c]: Looks like you have used spaces here instead of tabs.

**Comment by manuelVo** [comment]
My IDE switched defaults today. Seems like i missed that spot... It's fixed now.

**Comment by Gymnasiast** [comment]
Won't this require increasing the network version?

**Comment by manuelVo** [comment]
The callback isn't touched by the server or any other clients other than the one initiating the hire staff game command (which already exists and wasn't changed), so i don't see how this could cause problems. Is there any policy for when to increase the network version?

**Comment by Gymnasiast** [comment]
The policy is to increase the network version when a change will cause desyncs, or when we suspect it will.
I don't know a terrible lot about multiplayer, so I leave it to other dev team members to decide if it's required in this instance.

**Comment by janisozaur** [comment]
I checked that yesterday, I think it is fine to go in without change to network version. Older clients will simply not do callback/crash as in #1943, but that does not imply any desyncs. We will change it soon with @zaxcav's pathfinding changes anyway, so it will make sense to merge this before #4529, #4585 or #4576.

**Pull request merged**